File: raffia-core/src/main/java/com/codepoetics/raffia/streaming/FilteringWriter.java
package com.codepoetics.raffia.streaming;

import com.codepoetics.raffia.writers.BasketWriter;

public interface FilteringWriter<T extends BasketWriter<T>> extends BasketWriter<FilteringWriter<T>> {

  FilteringWriter<T> advance(T newTarget);
  T complete();

}

File: raffia-core/src/main/java/com/codepoetics/raffia/streaming/FilteringWriter.kt
package com.codepoetics.raffia.streaming

import com.codepoetics.raffia.writers.BasketWriter

interface FilteringWriter<T : BasketWriter<T>> : BasketWriter<FilteringWriter<T>> {

    fun advance(newTarget: T): FilteringWriter<T>
    fun complete(): T

}
